wpimath/src/main/native/cpp/geometry/Quaternion.cpp

```diff
@@ -36,19 +63,77 @@ bool Quaternion::operator==(const Quaternion& other) const {
   return std::abs(W() * other.W() + m_v.dot(other.m_v)) > 1.0 - 1E-9;
 }
 
-Quaternion Quaternion::Inverse() const {
+Quaternion Quaternion::Conjugate() const {
   return Quaternion{W(), -X(), -Y(), -Z()};
 }
 
+Quaternion Quaternion::Inverse() const {
+  double norm = Norm();
+  return Quaternion{W(), -X(), -Y(), -Z()} / (norm * norm);
+}
+
+double Quaternion::Norm() const {
+  return std::sqrt(W() * W() + m_v.dot(m_v));
+}
+
 Quaternion Quaternion::Normalize() const {
-  double norm = std::sqrt(W() * W() + X() * X() + Y() * Y() + Z() * Z());
+  double norm = Norm();
   if (norm == 0.0) {
     return Quaternion{};
   } else {
-    return Quaternion{W() / norm, X() / norm, Y() / norm, Z() / norm};
+    return Quaternion{W(), X(), Y(), Z()} / norm;
   }
 }
 
+Quaternion Quaternion::operator^(const double other) const {
```

Comment:
In retrospect this is typically used as some sort of exclusive or. Is this okay?

Reply:
Yea, `operator^` is generally not used for exponentiation. `Quaternion::Pow(double exp)` would work instead.